content/beginner/160_advanced-networking/secondary_cidr/eniconfig_crd.md

```diff
@@ -47,45 +48,95 @@ aws ec2 describe-subnets  --filters "Name=cidr-block,Values=100.64.*" --query 'S
 |  100.64.0.0/19  |  subnet-04f960ffc8be6865c  |  us-east-2b |
 +-----------------+----------------------------+-------------+
 {{< /output >}}
+
+Ensure new nodes are up and listed with 'Ready' status. 
+```
+kubectl get nodes  # Make sure new nodes are listed with 'Ready' status
+```
+{{ output }}
+NAME                                           STATUS   ROLES    AGE   VERSION
+ip-192-168-9-228.us-east-2.compute.internal     Ready    <none>   90m   v1.17.12-eks-7684af
+ip-192-168-71-211.us-east-2.compute.internal    Ready    <none>   92m   v1.17.12-eks-7684af
+ip-192-168-33-135.us-east-2.compute.internal    Ready    <none>   88m   v1.17.12-eks-7684af
+{{ /output }}
+ 
+{{% notice warning %}}
+ Wait till all three nodes show `Ready` status before moving to the next step.
+{{% /notice %}}
+ 
 Check your Worker Node SecurityGroup
 ```
 INSTANCE_IDS=(`aws ec2 describe-instances --query 'Reservations[*].Instances[*].InstanceId' --filters "Name=tag-key,Values=eks:cluster-name" "Name=tag-value,Values=eksworkshop*" --output text`)
-for i in "${INSTANCE_IDS[@]}"
-do
-  echo "SecurityGroup for EC2 instance $i ..."
-  aws ec2 describe-instances --instance-ids $i | jq -r '.Reservations[].Instances[].SecurityGroups[].GroupId'
-done  
+
+export NETCONFIG_SECURITY_GROUPS=$(for i in "${INSTANCE_IDS[@]}"; do  aws ec2 describe-instances --instance-ids $i | jq -r '.Reservations[].Instances[].SecurityGroups[].GroupId'; done  | sort | uniq | awk -vORS=, '{print $1 }' | sed 's/,$//')
+
+echo $NETCONFIG_SECURITY_GROUPS
 ```
 {{< output >}}
-SecurityGroup for EC2 instance i-03ea1a083c924cd78 ...
-sg-070d03008bda531ad
-sg-06e5cab8e5d6f16ef
-SecurityGroup for EC2 instance i-0a635aed890c7cc3e ...
-sg-070d03008bda531ad
-sg-06e5cab8e5d6f16ef
-SecurityGroup for EC2 instance i-048e5ec8815e5ea8a ...
 sg-070d03008bda531ad
 sg-06e5cab8e5d6f16ef
 {{< /output >}}
-Create custom resource **group1-pod-netconfig.yaml** for first subnet (100.64.0.0/19). Replace the SubnetId and SecuritGroupIds with the values from above. Here is how it looks with the configuration values for my environment
-
 Note: We are using same SecurityGroup for pods as your Worker Nodes but you can change these and use custom SecurityGroups for your Pod Networking
-
+ 
+Check the `yq` command runs successfully. Refer to `yq` setup in [Install Kubernetes Tools](https://www.eksworkshop.com/020_prerequisites/k8stools/)
+```
+yq help >/dev/null  && echo "yq command working" || "yq command not working"
 ```
+{{ output }}
+ yq command working
+{{ /output }}
+ 
+Create ENIConfig custom resources. One file per AZ. 
+```
+mkdir -p eniconfig
```

Comment:
Ensure which folder this reference is created in, or use the relative path from $HOME/~.